src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs

```diff
@@ -144,21 +168,35 @@ public void UpdateVariable(string name, FormulaValue value)
         /// <returns>The formula's result.</returns>
         public FormulaValue Eval(string expressionText, RecordValue parameters = null, ParserOptions options = null)
         {
-            return EvalAsync(expressionText, CancellationToken.None, parameters, options).Result;          
+            try
+            {
+                return EvalAsync(expressionText, CancellationToken.None, parameters, options).Result;
+            }
+            catch (RuntimeMaxCallDepthException)

```

Comment:
This is caught within evalAsync, it shouldn't reach here.

Reply:
Done